core/src/main/java/org/apache/calcite/sql/SqlOverOperator.java

```diff
@@ -115,6 +115,7 @@ public SqlOverOperator() {
 
     // Copied from validateOperands
     validator.setValidatedNodeType(call, ret);
+    // TODO: Should the type of the agg change just because it is an OVER operand?
```

Comment:
I removed the following line to work around [CALCITE-6126], but that broke other tests in ways I did not understand, so I left this comment. I'm happy to remove it if this is not best practice.

Reply:
Yes, we don't want to add TODOs. They never get done.

If you have a question, ask it in jira.